packages/@lwc/engine/dom/src/renderer.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { hasOwnProperty, isUndefined } from '@lwc/shared';
+
+import { Renderer } from '../../src';
+
+// TODO [#0]: Evaluate how we can extract the `$shadowToken$` property name in a shared package
+// to avoid having to synchronize it between the different modules.
+export const syntheticShadow = hasOwnProperty.call(Element.prototype, '$shadowToken$');
+
+export const renderer: Renderer<Node, Element> = {
+    syntheticShadow,
+
+    createElement(tagName: string, namespace: string): Element {
```

Comment:
why don't we just provide two methods? they have different signature and such...  I also don't see setAttributeNS, which is probably something that we are not supporting in compiler

Reply:
> why don't we just provide two methods?
My original goal was to keep the renderer surface as small as possible and avoid code duplication. If you have a strong opinion we can create 2 methods for each API supporting namespace.

>  I also don't see setAttributeNS, which is probably something that we are not supporting in compiler.
The `setAttributeNS` is used to set the `xlink:href` attribute from the compiler. The LightningElement class also exposes the `getAttributeNS`, `setAttributeNS`, `hasAttributeNS` and `removeAttributeNS`.